test/rewardsTest.js

```diff
@@ -0,0 +1,313 @@
+const chai = require("chai")
+const expect = chai.expect
+const { ethers, helpers } = require("hardhat")
+
+describe("Rewards", () => {
+  let alice
+  let bob
+  let carol
+
+  let rewards
+
+  beforeEach(async () => {
+    alice = 1
+    bob = 2
+    carol = 3
+
+    const RewardsStub = await ethers.getContractFactory("RewardsStub")
+    rewards = await RewardsStub.deploy()
+    await rewards.deployed()
+  })
+
+  describe("sortition pool rewards", async () => {
+    it("pays rewards proportionally", async () => {
+      await rewards.addOperator(alice, 10)
+      await rewards.addOperator(bob, 90)
+
+      const poolWeight = await rewards.getPoolWeight()
+
+      expect(poolWeight).to.equal(100)
+
+      await rewards.payReward(1000)
+
+      await rewards.withdrawRewards(alice)
+      const aliceRewards = await rewards.getWithdrawnRewards(alice)
+      await rewards.withdrawRewards(bob)
+      const bobRewards = await rewards.getWithdrawnRewards(bob)
+
+      expect(aliceRewards).to.be.equal(100)
+      expect(bobRewards).to.be.equal(900)
+    })
+
+    it("only pays to present members", async () => {
+      await rewards.addOperator(alice, 10)
+      await rewards.addOperator(bob, 90)
+
+      await rewards.updateOperatorWeight(bob, 0)
+
+      await rewards.payReward(1000)
+
+      await rewards.withdrawRewards(alice)
+      const aliceRewards = await rewards.getWithdrawnRewards(alice)
+      await rewards.withdrawRewards(bob)
+      const bobRewards = await rewards.getWithdrawnRewards(bob)
+
+      expect(aliceRewards).to.equal(1000)
+      expect(bobRewards).to.equal(0)
+    })
+
+    it("handles dust", async () => {
+      await rewards.addOperator(alice, 10)
+
+      await rewards.payReward(123)
+
+      const acc1 = await rewards.getGlobalAccumulator()
+      expect(acc1).to.equal(12)
+
+      const dust1 = await rewards.getRoundingDust()
+      expect(dust1).to.equal(3)
+
+      await rewards.addOperator(bob, 20)
+
+      await rewards.payReward(987)
+
+      const acc2 = await rewards.getGlobalAccumulator()
+      expect(acc2).to.equal(45)
+
+      const dust2 = await rewards.getRoundingDust()
+      expect(dust2).to.equal(0)
+
+      await rewards.withdrawRewards(alice)
+      const aliceRewards = await rewards.getWithdrawnRewards(alice)
+
+      expect(aliceRewards).to.equal(450)
+
+      await rewards.withdrawRewards(bob)
+      const bobRewards = await rewards.getWithdrawnRewards(bob)
+
+      expect(bobRewards).to.equal(660)
+    })
+
+    it("handles sequences", async () => {
+      await rewards.addOperator(alice, 10)
+      await rewards.addOperator(bob, 90)
+
+      // alice: 100; bob: 900
+      await rewards.payReward(1000)
+
+      const globalAcc1 = await rewards.getGlobalAccumulator()
+      expect(globalAcc1).to.equal(10)
+
+      await rewards.updateOperatorWeight(bob, 0)
+
+      const bobRew1 = await rewards.getAccruedRewards(bob)
+      expect(bobRew1).to.equal(900)
+
```

Comment:
For completeness, should we assert `getAccruedRewards` of Alice?

Reply:
👍